vms/platformvm/warp/benchmarks/benchmarks_test.go

```diff
@@ -0,0 +1,478 @@
+		getValidatorsOutput := make(map[ids.NodeID]*validators.GetValidatorOutput)
+		for i := 0; i < valCounts; i++ {
+			validator := getValidatorOutputs[i]
+			getValidatorsOutput[validator.NodeID] = validator
+		}
+		validatorState := &validators.TestState{
+			GetValidatorSetF: func(context.Context, uint64, ids.ID) (map[ids.NodeID]*validators.GetValidatorOutput, error) {
+				return getValidatorsOutput, nil
+			},
+		}
+
+		b.Run(fmt.Sprintf("%d", valCounts), func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				vals, _, err := warp.GetCanonicalValidatorSet(
+					context.Background(),
+					validatorState,
+					pChainHeight,
+					subnetID,
+				)
+				require.NoError(err)
+				require.Len(vals, valCounts)
+			}
+		})
+	}
+}
+
+func BenchmarkGetCanonicalValidatorSetByDepth(b *testing.B) {
+	require := require.New(b)
+	var (
+		validatorsCount     = 10_000
+		heightsRange        = uint64(20)
+		validatorsPerHeight = validatorsCount / int(heightsRange)
+
+		subnetID            = ids.GenerateTestID()
+		getValidatorOutputs = make(map[uint64](map[ids.NodeID]*validators.GetValidatorOutput))
+	)
+
+	for depth := uint64(0); depth < heightsRange; depth++ {
+		getValidatorOutputs[heightsRange-depth] = make(map[ids.NodeID]*validators.GetValidatorOutput)
+		for i := 0; i < validatorsPerHeight; i++ {
+			nodeID := ids.GenerateTestNodeID()
+			blsPrivateKey, err := bls.NewSecretKey()
+			require.NoError(err)
+			blsPublicKey := bls.PublicFromSecretKey(blsPrivateKey)
+			getValidatorOutputs[heightsRange-depth][nodeID] = &validators.GetValidatorOutput{
+				NodeID:    nodeID,
+				PublicKey: blsPublicKey,
+				Weight:    20,
+			}
+		}
+	}
+	validatorState := &validators.TestState{
+		GetValidatorSetF: func(_ context.Context, depth uint64, subnetID ids.ID) (map[ids.NodeID]*validators.GetValidatorOutput, error) {
+			res, found := getValidatorOutputs[depth]
+			if !found {
+				return nil, database.ErrNotFound
+			}
+			return res, nil
+		},
+	}
+
+	b.StartTimer() // start testing
+	for depth := uint64(0); depth < heightsRange; depth++ {
+		b.Run(fmt.Sprintf("%d", depth), func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				vals, _, err := warp.GetCanonicalValidatorSet(
+					context.Background(),
+					validatorState,
+					heightsRange-depth,
+					subnetID,
+				)
+				require.NoError(err)
+				require.Len(vals, validatorsPerHeight)
+			}
+		})
+	}
+	b.StopTimer() // done testing
+}
+
+func BenchmarkGetValidatorSetBySize_Validators_100(b *testing.B) {
+	benchGetValidatorSetBySize(b, 100)
+}
+
+func BenchmarkGetValidatorSetBySize_Validators_1000(b *testing.B) {
+	benchGetValidatorSetBySize(b, 1000)
+}
+
+func BenchmarkGetValidatorSetBySize_Validators_10000(b *testing.B) {
+	benchGetValidatorSetBySize(b, 10_000)
+}
+
+func benchGetValidatorSetBySize(b *testing.B, validatorsCount int) {
+	b.StopTimer()
+	require := require.New(b)
+
+	// Prepare the bench environment (don't time it)
+	subnetID := ids.GenerateTestID()
+	env := newEnvironment(b, subnetID)
```

Comment:
This will create a new environment where `subnetID` is tracked, such that the validator set gets cached.

It seems like this benchmark will actually hit the cache every single time after the first iteration, such that it does not actually test what we want.

Could we fix this by making the size of the cache parametrizable, so we can set it to 0 and avoid hitting the cache?

Another alternative would be to set up a benchmark where we request the validator set for a subnet that is NOT tracked, so that it is not kept in the cache, but that seems overly special cased imo.